bson/bson_binary_vector_spec_test.go

```diff
@@ -0,0 +1,198 @@
+		require.NoErrorf(t, err, "reading test file %s", filepath)
+
+		var tests bsonBinaryVectorTests
+		require.NoErrorf(t, json.Unmarshal(content, &tests), "parsing test file %s", filepath)
+
+		t.Run(tests.Description, func(t *testing.T) {
+			t.Parallel()
+
+			for _, test := range tests.Tests {
+				test := test
+				t.Run(test.Description, func(t *testing.T) {
+					t.Parallel()
+
+					runBsonBinaryVectorTest(t, tests.TestKey, test)
+				})
+			}
+		})
+	}
+
+	t.Run("Insufficient vector data FLOAT32", func(t *testing.T) {
+		t.Parallel()
+
+		val := Binary{Subtype: TypeBinaryVector}
+
+		for _, tc := range [][]byte{
+			{byte(Float32Vector), 0, 42},
+			{byte(Float32Vector), 0, 42, 42},
+			{byte(Float32Vector), 0, 42, 42, 42},
+
+			{byte(Float32Vector), 0, 42, 42, 42, 42, 42},
+			{byte(Float32Vector), 0, 42, 42, 42, 42, 42, 42},
+			{byte(Float32Vector), 0, 42, 42, 42, 42, 42, 42, 42},
+		} {
+			t.Run(fmt.Sprintf("marshaling %d bytes", len(tc)-2), func(t *testing.T) {
+				val.Data = tc
+				b, err := Marshal(D{{"vector", val}})
+				require.NoError(t, err, "marshaling test BSON")
+				var got struct {
+					Vector Vector
+				}
+				err = Unmarshal(b, &got)
+				require.ErrorContains(t, err, errInsufficientVectorData.Error())
+			})
+		}
+	})
+
+	t.Run("Padding specified with no vector data PACKED_BIT", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("Marshaling", func(t *testing.T) {
+			_, err := NewPackedBitVector(nil, 1)
+			require.EqualError(t, err, errNonZeroVectorPadding.Error())
+		})
+		t.Run("Unmarshaling", func(t *testing.T) {
+			val := D{{"vector", Binary{Subtype: TypeBinaryVector, Data: []byte{byte(PackedBitVector), 1}}}}
+			b, err := Marshal(val)
+			require.NoError(t, err, "marshaling test BSON")
+			var got struct {
+				Vector Vector
+			}
+			err = Unmarshal(b, &got)
+			require.ErrorContains(t, err, errNonZeroVectorPadding.Error())
+		})
+	})
+
+	t.Run("Exceeding maximum padding PACKED_BIT", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("Marshaling", func(t *testing.T) {
+			_, err := NewPackedBitVector(nil, 8)
+			require.EqualError(t, err, errVectorPaddingTooLarge.Error())
+		})
+		t.Run("Unmarshaling", func(t *testing.T) {
+			val := D{{"vector", Binary{Subtype: TypeBinaryVector, Data: []byte{byte(PackedBitVector), 8}}}}
+			b, err := Marshal(val)
+			require.NoError(t, err, "marshaling test BSON")
+			var got struct {
+				Vector Vector
+			}
+			err = Unmarshal(b, &got)
+			require.ErrorContains(t, err, errVectorPaddingTooLarge.Error())
+		})
+	})
+}
+
+func convertSlice[T int8 | float32 | byte](s []interface{}) []T {
+	v := make([]T, len(s))
+	for i, e := range s {
+		f := math.NaN()
+		switch v := e.(type) {
+		case float64:
+			f = v
+		case string:
+			if v == "inf" {
+				f = math.Inf(0)
+			} else if v == "-inf" {
+				f = math.Inf(-1)
+			}
+		}
+		v[i] = T(f)
+	}
+	return v
+}
+
+func runBsonBinaryVectorTest(t *testing.T, testKey string, test bsonBinaryVectorTestCase) {
+	if !test.Valid {
```

Comment:
Since we've unified vector and packed bit vector, we don't want to skip invalid tests.